shared/middleware/files_api.rb

```diff
@@ -783,7 +783,19 @@ def files_put_file(encrypted_channel_id, filename, body)
   # Read a metadata file, caching the result for 1 hour.
   #
   get %r{/v3/files-public/([^/]+)/.metadata/([^/]+)$} do |encrypted_channel_id, filename|
-    file = get_file('files', encrypted_channel_id, "#{METADATA_PATH}/#{filename}")
+    s3_prefix = "#{METADATA_PATH}/#{filename}"
+    if THUMBNAIL_FILENAME == filename
+      temp_url = FileBucket.new.make_temporary_public_url(encrypted_channel_id, s3_prefix)
+      rating = ImageModeration.rate_image(temp_url)
+      if %i(adult racy).include? rating
+        new_score = StorageApps.new(storage_id('user')).increment_abuse(encrypted_channel_id)
+        FileBucket.new.replace_abuse_score(encrypted_channel_id, s3_prefix, new_score)
```

Comment:
I'm going to punt on opening a Zendesk ticket, that's not something that came up in earlier discussion of this feature although I do think some notification is appropriate.  I'll consider that follow-up work.

I didn't know about `ReportAbuseController` though, I'll see if I can get it reporting on all related files now.

Reply:
Started work on integrating with `ReportAbuseController` when I learned that there's a bug in that controller - it doesn't flag assets in the AnimationBucket - and no tests over it either.  So I've started work on adding tests and fixing that problem but it's going to take a while, because it's a pegasus-dashboard cross-cutting concern being tested from a Rails controller test and seems to require calling Rack internals to set up the appropriate test scenario :cry:.

All that said, I think we could just merge this PR and get things up and running - reporting abuse on the project and thumbnail is sufficient for our MVP of not serving adult content in the public gallery.